### ISSUE TITLE
Don't get a proper QuickLook preview, just a window with DEBUG as a header

### ISSUE THREAD

**Issue by CameronBanga** [mherrmann/QuickLook#2]
Loving fman so far, just installed this plugin, but not getting any luck with QuickLook. Every time I try to use it, I get the following:

![screen shot 2018-01-03 at 11 50 31 am](https://user-images.githubusercontent.com/611354/34532623-6cbef72c-f07c-11e7-8eec-267decdd5c96.png)

**Comment by CameronBanga** [comment]
Seeing this for PNGs, JPEGs, any sort of file I'd normally get a preview with in Finder.

**Comment by mherrmann** [comment]
Thank you for the kind words! You're right, I am experiencing this as well. I suspect it's a problem with High Sierra. Let me check and get back to you.

**Comment by mherrmann** [comment]
Should be fixed. Can you re-install the plugin and let me know if it works for you now?

**Comment by CameronBanga** [comment]
@mherrmann Works now! Fantastic. Definitely not regretting buying that license earlier today now. Great work, excited to see how fman progresses!

**Comment by mherrmann** [comment]
Glad to hear it! Thank you very much for getting a license. I really appreciate it.

fman is just getting started ;)

**Comment by CameronBanga** [comment]
Awesome, just saw it on HackerNews last week I think, and was out sick for a few days and finally getting around to playing around. Works almost exactly with how I imagine/use the file system, so very excited to see it grow.

**Comment by mherrmann** [comment]
Cool :) If you find there is something that needs to be improved, just let me know.

**Comment by mwean** [comment]
I'm getting this same issue on Sierra, but it's opening in a separate app (qlmanage)

**Comment by mherrmann** [comment]
@mwean could you post a screenshot?

**Comment by mwean** [comment]
Here's the quicklook window:
![screenshot2](https://user-images.githubusercontent.com/732173/37719060-91218ea8-2cfa-11e8-8399-17e5785fcc82.png)

And this is the application:
<img width="204" alt="screenshot" src="https://user-images.githubusercontent.com/732173/37719106-a7496a98-2cfa-11e8-96f5-193cf94272a4.png">

**Comment by mherrmann** [comment]
@mwean do you get the same when you type

    qlmanage -p "/path/to/Migration Steps.txt"

?

Also, does it happen for all files? Eg., have you tried for images?

**Comment by mwean** [comment]
Yeah, it seems the same if I use that command. It seems to work better for images, but it still shows the qlmanage icon in the dock and the window has "[DEBUG]" in the name.

I also got this when I ran the command:

```
2018-03-22 10:42:01.635 qlmanage[46158:10645282] *** CFMessagePort: bootstrap_register(): failed 1100 (0x44c) 'Permission denied', port = 0xa303, name = 'com.apple.coredrag'
See /usr/include/servers/bootstrap_defs.h for the error codes.
2018-03-22 10:42:01.745 qlmanage[46158:10645282] *** CFMessagePort: bootstrap_register(): failed 1100 (0x44c) 'Permission denied', port = 0xce3b, name = 'com.apple.tsm.portname'
See /usr/include/servers/bootstrap_defs.h for the error codes.
2018-03-22 10:42:01.745 qlmanage[46158:10645282] *** CFMessagePort: bootstrap_register(): failed 1100 (0x44c) 'Permission denied', port = 0xd103, name = 'com.apple.CFPasteboardClient'
See /usr/include/servers/bootstrap_defs.h for the error codes.
```
So it's possible there's other stuff going on.

**Comment by mherrmann** [comment]
Interesting, thanks Matt. Does it work when you do

    sudo qlmanage ...

?

**Comment by mwean** [comment]
That seems to work better (in that it pops up right away), but I still get those error messages, and it still says "[DEBUG]"

**Comment by mherrmann** [comment]
So you're seeing the correct contents inside the Preview window then? The "[DEBUG]" is ugly but the error message don't matter too much I think, because you wouldn't see them when launching the feature from fman.

**Comment by mwean** [comment]
Sorry, yes I see the correct content with sudo, not in fman

**Comment by mherrmann** [comment]
If I recall correctly, it's somehow possible to write plugins for QuickLook. Do you happen to know of a plugin that might be installed on your system - and may require the elevated permissions?

**Comment by mwean** [comment]
Yeah, I definitely have plugins, but they all work fine in other apps like PathFinder and ForkLift. Are you invoking quicklook in a different way?

**Comment by mherrmann** [comment]
No, I'm doing exactly `qlmanage -p "/path/to/Migration Steps.txt"`. Hm...

**Comment by mwean** [comment]
No, I'm saying are you doing something different than other Mac apps

**Comment by mherrmann** [comment]
I don't know how the other apps do it, so I can't really answer that question unfortunately. One thing that's special about fman is that it is a cross-platform Qt app, rather than a native one. I'm just trying to "hack" the other apps to see how they're doing it.

**Comment by montemartin** [comment]
I hate to reopen a closed ticket, but this is definitely still an issue, and a big one for me. It's the biggest barrier to me using fman exclusively. My typical use for a file browser is to quickly traverse folders and get a quick glimpse at the contents of the files in them. Having to open them in an app steals keyboard focus and defeats the purpose of fman. 

It works fine for some file types, but hangs one others. It seems to work fine on .sh .config and  files with no extension. It hangs on .html, .php, .py, any image/video files, and hangs very badly on .json files.

You asked previously about QuickLook extensions so I'll point you to this repo https://github.com/sindresorhus/quick-look-plugins I have several of these installed and they work fine, even when I'm using preview via fman. I've tried uninstalling all of them and still have the same issues.

Hope this is helpful. I'd love to get this working correctly.
